packages/nextjs/components/scaffold-eth/BatchStatusIndicators.tsx

```diff
@@ -0,0 +1,71 @@
+"use client";
+
+import { zeroAddress } from "viem";
+import { useAccount } from "wagmi";
+import { CheckCircleIcon, UserGroupIcon, XCircleIcon } from "@heroicons/react/24/solid";
+import { useScaffoldReadContract } from "~~/hooks/scaffold-eth";
+
+export const BatchStatusIndicators = () => {
+  const { address: userAddress } = useAccount();
+
+  // Check if address is in allowList (batch member)
+  const { data: allowListStatus, isLoading: isLoadingAllowList } = useScaffoldReadContract({
+    contractName: "BatchRegistry",
+    functionName: "allowList",
+    args: [userAddress ?? zeroAddress],
+    chainId: 42161,
+  });
+
+  // Check if address has checked in
+  const { data: contractAddress, isLoading: isLoadingContractAddress } = useScaffoldReadContract({
+    contractName: "BatchRegistry",
+    functionName: "yourContractAddress",
+    args: [userAddress ?? zeroAddress],
+    chainId: 42161,
+  });
+
+  const isInBatch = allowListStatus !== undefined ? !!allowListStatus : null;
+  const isCheckedIn = contractAddress !== undefined ? contractAddress !== zeroAddress : null;
```

Comment:
These 2 variables should be purely booleans, so they should never be `null`, only `true` or `false`.
And we don't need ternary operators here. It's more difficult to understand than simply using logical operators.